tests/test_redis_pytest.py

```diff
@@ -1,146 +1,212 @@
 """Tests the pytest-redis with a running redis instance.
 
 Tests should be launched from the root directory with:
-py.test  --redis-port=<port> --redis-host=<host> --redis-list-key=<list_to_use>
+py.test --redis-port=<port> --redis-host=<host> --redis-list-key=<list_to_use>
 
 """
+import pytest_redis
+import threading
+import os.path
 
 
-def test_consumption_of_item(testdir, redis_args, redis_connection):
-    """Make sure that we can consume multiple items."""
-    [redis_connection.lpush(redis_args['redis-list-key'],
-                            "test_consumption_of_item.py::test_run" + str(i))
-        for i in range(5)]
+def default_pytest_redis_args():
+    """Return default options for each pytest execution."""
+    return ['-v', '-p', 'pytest_redis']
 
-    # create a temporary pytest test module
-    testdir.makepyfile("""
-        def test_run0():
-            assert True
 
-        def test_run1():
-            assert True
+def get_option_array(option_dict):
+    """Return cmdline options for a dict in '--key=val' form."""
+    return ["--{}={}".format(k, v) for k, v in option_dict.items()]
 
-        def test_run2():
-            assert True
 
-        def test_run3():
-            assert True
+def create_test_file(testdir, filename, text):
+    """Create test file with the given name and text contents."""
+    the_kwargs = {
+        filename: text
+    }
+    ext = ""
+    try:
+        ext = filename.split(".")[-1]
+    except:
+        pass
+
+    testdir.makefile(ext, **the_kwargs)
+
+
+def get_standard_args(option_dict):
+    """Returns standard pytest redis_args combinbed with option dict"""
+    return default_pytest_redis_args() + get_option_array(option_dict)
+
 
-        def test_run4():
+def create_test_dir(testdir, dirname):
+    """Create test file with the given name and text contents."""
+    testdir.mkdir(dirname)
+
+
+def setup_multiple_consumer_threads(testdir, py_test_args, num_threads):
+    """Return multiple threads that will run pytest with the given args."""
+    def run_consumer(testdir, test_args):
+        result = testdir.runpytest(*test_args)
+        assert result.ret != pytest_redis.EXIT_INTERRUPTED
+        assert result.ret != pytest_redis.EXIT_TESTSFAILED
+        assert result.ret == pytest_redis.EXIT_OK
```

Comment:
`EXIT_OK` assumes that we've collected and run at least one test, I assume, but since we play with the return value can you also check here to make sure that at least one test has been collected and run?